.changeset/lazy-colts-help.md

```diff
@@ -0,0 +1,56 @@
+---
+'@backstage/cli': patch
+'@backstage/core-app-api': patch
+'@backstage/core-components': patch
+'@backstage/core-plugin-api': patch
+'@backstage/create-app': patch
+'@backstage/dev-utils': patch
+'@backstage/integration-react': patch
+'@backstage/test-utils': patch
+'@backstage/theme': patch
+'@backstage/plugin-api-docs': patch
+'@backstage/plugin-badges': patch
+'@backstage/plugin-bitrise': patch
+'@backstage/plugin-catalog': patch
+'@backstage/plugin-catalog-import': patch
+'@backstage/plugin-catalog-react': patch
+'@backstage/plugin-circleci': patch
+'@backstage/plugin-cloudbuild': patch
+'@backstage/plugin-code-coverage': patch
+'@backstage/plugin-config-schema': patch
+'@backstage/plugin-cost-insights': patch
+'@backstage/plugin-explore': patch
+'@backstage/plugin-fossa': patch
+'@backstage/plugin-gcp-projects': patch
+'@backstage/plugin-git-release-manager': patch
+'@backstage/plugin-github-actions': patch
+'@backstage/plugin-github-deployments': patch
+'@backstage/plugin-gitops-profiles': patch
+'@backstage/plugin-graphiql': patch
+'@backstage/plugin-ilert': patch
+'@backstage/plugin-jenkins': patch
+'@backstage/plugin-kafka': patch
+'@backstage/plugin-kubernetes': patch
+'@backstage/plugin-lighthouse': patch
+'@backstage/plugin-newrelic': patch
+'@backstage/plugin-org': patch
+'@backstage/plugin-pagerduty': patch
+'@backstage/plugin-register-component': patch
+'@backstage/plugin-rollbar': patch
+'@backstage/plugin-scaffolder': patch
+'@backstage/plugin-search': patch
+'@backstage/plugin-sentry': patch
+'@backstage/plugin-shortcuts': patch
+'@backstage/plugin-sonarqube': patch
+'@backstage/plugin-splunk-on-call': patch
+'@backstage/plugin-tech-radar': patch
+'@backstage/plugin-techdocs': patch
+'@backstage/plugin-todo': patch
+'@backstage/plugin-user-settings': patch
+'@backstage/plugin-welcome': patch
+'@backstage/plugin-xcmetrics': patch
+---
+
+- Bumping `material-ui/core` version to at least `4.12.2` as they made some breaking changes in later versions which broke Pagination of the Table.
+
+- Switching out `material-table` to `@material-table/core` for support for the later versions of `material-ui/core`
```

Comment:
NIT: How about mentioning that the `onChangePage` and `onChangeRowsPerPage` properties have been renamed in the `Table` component?

Reply:
Good shout, I purposefully left this out as our Table API has stayed the same it's only if using `material-table` directly. But I can add something to the changeset stating this.

Reply:
I think next release we will add support for both props, and then remove the old one and do a proper deprecation like material-ui should have done :shrug:

Reply:
> Good shout, I purposefully left this out as our Table API has stayed the same it's only if using material-table directly. But I can add something to the changeset stating this.

I'm not sure about that. Didn't you update the tables in the circleci, bitrise, and cloudbuild plugins that all use `import { Table } from '@backstage/core-components`?

Reply:
Hmm good catch - The `api-reports` just re-exports the props, so when I just scanned to find what I'd done again I didn't see it changing the props. https://github.com/backstage/backstage/pull/6581/files?file-filters%5B%5D=.md#diff-ea569eb8425f004ed49a86194d7b5b19b5dff946ced5c6f77933d48edbb0f036R29